code/scripting/api/objs/enums.h

```diff
@@ -7,148 +7,182 @@
-const int32_t LE_MESSAGE_PRIORITY_HIGH       = 73;
-const int32_t LE_OPTION_TYPE_SELECTION       = 78;
-const int32_t LE_OPTION_TYPE_RANGE           = 79;
-const int32_t LE_ASF_EVENTMUSIC              = 80;
-const int32_t LE_ASF_MENUMUSIC               = 81;
-const int32_t LE_ASF_VOICE                   = 82;
-const int32_t LE_CONTEXT_VALID               = 83;
-const int32_t LE_CONTEXT_SUSPENDED           = 84;
-const int32_t LE_CONTEXT_INVALID             = 85;
-const int32_t LE_FIREBALL_MEDIUM_EXPLOSION   = 86;
-const int32_t LE_FIREBALL_LARGE_EXPLOSION    = 87;
-const int32_t LE_FIREBALL_WARP_EFFECT        = 88;
-const int32_t LE_GR_RESIZE_NONE              = 89; // the sequence and offsets of the LE_GR_* #defines should correspond to the GR_* #defines
-const int32_t LE_GR_RESIZE_FULL              = 90;
-const int32_t LE_GR_RESIZE_FULL_CENTER       = 91;
-const int32_t LE_GR_RESIZE_MENU              = 92;
-const int32_t LE_GR_RESIZE_MENU_ZOOMED       = 93;
-const int32_t LE_GR_RESIZE_MENU_NO_OFFSET    = 94;
-const int32_t LE_TBOX_FLASH_NAME             = 95;
-const int32_t LE_TBOX_FLASH_CARGO            = 96;
-const int32_t LE_TBOX_FLASH_HULL             = 97;
-const int32_t LE_TBOX_FLASH_STATUS           = 98;
-const int32_t LE_TBOX_FLASH_SUBSYS           = 99;
-const int32_t LE_LUAAI_ACHIEVABLE			 = 100;
-const int32_t LE_LUAAI_NOT_YET_ACHIEVABLE	 = 101;
-const int32_t LE_LUAAI_UNACHIEVABLE			 = 102;
-const int32_t LE_MOVIE_PRE_FICTION           = 103; // the sequence and offsets of the LE_MOVIE_* #defines should correspond to the MOVIE_* #defines
-const int32_t LE_MOVIE_PRE_CMD_BRIEF         = 104;
-const int32_t LE_MOVIE_PRE_BRIEF             = 105;
-const int32_t LE_MOVIE_PRE_GAME              = 106;
-const int32_t LE_MOVIE_PRE_DEBRIEF           = 107;
-const int32_t LE_MOVIE_POST_DEBRIEF          = 108;
-const int32_t LE_MOVIE_END_CAMPAIGN          = 109;
-const int32_t LE_SCORE_BRIEFING              = 110; // the sequence and offsets of the LE_SCORE_* #defines should correspond to the SCORE_* #defines
-const int32_t LE_SCORE_DEBRIEFING_SUCCESS    = 111;
-const int32_t LE_SCORE_DEBRIEFING_AVERAGE    = 112;
-const int32_t LE_SCORE_DEBRIEFING_FAILURE    = 113;
-const int32_t LE_SCORE_FICTION_VIEWER        = 114;
-const int32_t LE_NOT_YET_PRESENT             = 115; // the sequence and offsets of these three #defines should correspond to the ShipStatus enums
-const int32_t LE_PRESENT                     = 116;
-const int32_t LE_EXITED                      = 117;
-const int32_t LE_DC_IS_HULL                  = (1<<0); // the following definitions are bitfields independent of the usual indexing
-const int32_t LE_DC_VAPORIZE                 = (1<<1);
-const int32_t LE_DC_SET_VELOCITY             = (1<<2);
-const int32_t LE_DC_FIRE_HOOK                = (1<<3);
-
-const int ENUM_NEXT_INDEX = 118; // <<<<<<<<<<<<<<<<<<<<<<
-extern flag_def_list Enumerations[];
-extern size_t Num_enumerations;
+enum lua_enum : int32_t {
```

Comment:
should this be `enum class`?

Reply:
Yes and no.
While scoping it in an ``enum class`` certainly makes sense and would clean up the global namespace, it also comes with some downsides.
Notably, since it wasn't scoped before, scoping it now would mean that every usage of lua enum values in every file needs to be touched and adjusted for the new scope. With every enum being used once to thrice, that's a lot of additional line bloat in this PR.
But it certainly would be possible, if really wanted

Reply:
Ah, good point.  I'm fine with leaving it as-is.